contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -35,94 +34,79 @@ interface ILinkAvailable {
   /// @param minBalances the list of corresponding minBalance for the target address
-  function setWatchList(address[] calldata addresses, uint256[] calldata minBalances) external onlyOwner {
-    if (addresses.length != minBalances.length) {
+  /// @param topUpAmounts the list of corresponding minTopUp for the target address
+  function setWatchList(
+    address[] calldata addresses,
+    uint96[] calldata minBalances,
+    uint96[] calldata topUpAmounts
+  ) external onlyOwner {
+    if (addresses.length != minBalances.length || addresses.length != topUpAmounts.length) {
       revert InvalidWatchList();
     }
-    // first, remove all existing addresses from list
-    for (uint256 idx = s_watchList.length(); idx > 0; idx--) {
-      (address target, ) = s_watchList.at(idx - 1);
-      require(s_watchList.remove(target), "LinkAvailableBalanceMonitor: unable to setWatchlist");
-    }
-    // then set new addresses
-    for (uint256 idx = 0; idx < addresses.length; idx++) {
-      if (s_watchList.contains(addresses[idx])) {
-        revert DuplicateAddress(addresses[idx]);
-      }
-      if (addresses[idx] == address(0)) {
-        revert InvalidWatchList();
-      }
-      s_watchList.set(addresses[idx], minBalances[idx]);
-    }
-    emit WatchlistUpdated();
-  }
-
-  /// @notice Adds addresses to the watchlist without overwriting existing members
-  /// @param addresses the list of target addresses to watch (could be direct target or IAggregatorProxy)
-  /// @param minBalances the list of corresponding minBalance for the target address
-  function addToWatchList(address[] calldata addresses, uint256[] calldata minBalances) external onlyOwner {
-    if (addresses.length != minBalances.length) {
-      revert InvalidWatchList();
+    for (uint256 idx = 0; idx < s_watchList.length; idx++) {
+      delete s_targets[s_watchList[idx]];
     }
     for (uint256 idx = 0; idx < addresses.length; idx++) {
-      if (s_watchList.contains(addresses[idx])) {
+      address targetAddress = addresses[idx];
+      if (s_targets[targetAddress].isActive) {
         revert DuplicateAddress(addresses[idx]);
       }
       if (addresses[idx] == address(0)) {
         revert InvalidWatchList();
       }
-      s_watchList.set(addresses[idx], minBalances[idx]);
-    }
-    emit WatchlistUpdated();
-  }
-
-  /// @notice Removes addresses from the watchlist
-  /// @param addresses the list of target addresses to remove from the watchlist
-  function removeFromWatchlist(address[] calldata addresses) external onlyOwner {
-    for (uint256 idx = 0; idx < addresses.length; idx++) {
-      if (!s_watchList.contains(addresses[idx])) {
-        revert InvalidWatchList();
-      }
-      s_watchList.remove(addresses[idx]);
+      s_targets[targetAddress] = MonitoredAddress({
+        isActive: true,
+        minBalance: minBalances[idx],
+        topUpAmount: topUpAmounts[idx],
+        lastTopUpTimestamp: 0
+      });
```

Comment:
should we check `topUpAmount > 0` here? I think if we enforce that here we can remove all the `if topUpAmount > 0` checks elsewhere